config/impl-base/src/test/java/org/jboss/arquillian/config/impl/extension/ClasspathReplacementInArqXmlTestCase.java

```diff
@@ -0,0 +1,59 @@
+package org.jboss.arquillian.config.impl.extension;
+
+import java.net.URL;
+import org.jboss.arquillian.config.descriptor.impl.AssertXPath;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.jboss.arquillian.config.descriptor.impl.AssertXPath.assertXPath;
+
+public class ClasspathReplacementInArqXmlTestCase extends AbstractReplacementInArqXmlTestBase {
+
+    /**
+     * Name of the arquillian.xml to test
+     */
+    private static final String NAME_ARQ_XML = "arquillian_classpathprop.xml";
+
+    /**
+     * Name of the system property for EL expressions
+     */
+    private static final String SYSPROP_ARQ_CONTAINER = "arquillian.container";
+
+    private static final String VALUE_EL_OVERRIDE = "ALR";
+
+    /**
+     * Sets the name of the arquillian.xml under test
+     */
+    @BeforeClass
```

Comment:
both @BeforeClass and @AfterClass methods are almost same (including the variables), shouldn't that be moved to the template as well? I know that the arquillian.xml file name is different - that can be retrieved by an abstract method `getArquillianXmlFileName()`

Reply:
I understand what you mean and I will do but I always have the feeling that using abstractions in tests makes look tests somehow less readable in the sense of just checking the class and know what they are doing (you need to navigate through parent classes and be present about that), but I know it is something personal.

Reply:
In fact, this approach you mention does not work because `@BeforeAll` is (and must) be static so we cannot retrieve it from an abstract method.

Reply:
Yeah, right. You know, it's Friday and my brain is probably in the weekend mode :-D